### PR TITLE
Update sanitations.bal and operationId.bal for OData V4 Open API Specifications

### PR DESCRIPTION
**Pull request by Osara-B**
## Purpose
Updating the sanitations.bal and operationId.bal to sanitize OData V4 Open API specifications. Adding a new example for ODATAV4 to the README.md.

## Checklist
- [ ] Updated sanitations.bal to correctly sanitize ODATAV4 Open API Specifications
- [ ] Updated operationId.bal to correctly sanitize ODATAV4 Open API Specifications
- [ ] Added a new example in README.md demonstrating the updated sanitization process for OData V4 Open API specifications.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# SAP S/4HANA OpenAPI Specification Sanitization
+
+_Author_: @niveathika \
```
Comment (niveathika) [inline docs/README.md]:
```suggestion
_Author_: @Osara-B  \  
_Reviewer_: @niveathika 
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+_Created_: 2024/05/21 \
+_Updated_: 2024/05/21 \
```
Comment (niveathika) [inline docs/README.md]: Dates need to be updated

File: docs/sanitation/operationId.bal [hunk 3]
```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (niveathika) [inline docs/sanitation/operationId.bal:1]: Format the code. Same for all files

**Review by niveathika — CHANGES_REQUESTED** [review]

File: docs/sanitation/operationId.bal [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+public function main(string apiName) returns error? {
+    string specPath = string `spec/${apiName}.json`;
+    json openAPISpec = check io:fileReadJson(specPath);
+
+    Specification spec = check openAPISpec.cloneWithType(Specification);
+
+    boolean isODATA4 = false;
+    if (spec.x\-sap\-api\-type=="ODATAV4"){
```
Comment (niveathika) [inline docs/sanitation/operationId.bal]: Remove brackets in the condition

File: docs/sanitation/operationId.bal (hunk 3 above)
Comment (Osara-B) [reply docs/sanitation/operationId.bal:1]: Changes done in the commit [Format code files and refresh timestamps and author details](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/2/commits/d418fc6eb12fe437df297866a4746d8e2eb234c5)

> Format the code. Same for all files

**Review by niveathika — COMMENTED** [review]
@Osara-B's changes looks good to me. Lets wait for @osanda2024's fixes too and merge

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
+# SAP S/4HANA OpenAPI Specification Sanitization
+
+_Author_: @Osara-B  \  
+_Reviewer_: @niveathika 
+_Created_: 2024/06/27 \
+_Updated_: 2024/06/27 \
+_Edition_: Swan Lake
```
Comment (niveathika) [inline docs/README.md]:
```suggestion
_Author_: @Osara-B @osanda2024 \
_Reviewer_: @niveathika \
_Created_: 2024/06/27 \
_Updated_: 2024/06/28 \
_Edition_: Swan Lake  
```

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Move inline enum parameters to schemas. This simplifies function definitions and enhances documentation. Schema names
+   are generated based on the following pattern:
+   `${The Resource Name}Of${Base Path Name}`
+
+2. Remove unnecessary grouping prefixes from schema names. For example:
+   `Api_sales_contract_srvA_Salesorder` -> `A_Salesorder`
+   `com\.sap\.gateway\.srvd_a2x\.api_defect\.v0001\.Defect_Type` -> `Defect`
+
+3. Add operation Ids. This is more user-friendly with SAP-specific scripts. The logic for parameter sanitization is
```
Comment (niveathika) [inline docs/README.md]:
```suggestion
3. Improve response schema names by removing unnecessary prefixes and suffixes and renaming them to be more descriptive.
   `wrapper` -> `A_InspectionlotWrapper`
   `Collection of A_InspectionlotType` -> `CollectionOfA_Inspectionlot`

4. Add operation Ids. This is more user-friendly with SAP-specific scripts. The logic for parameter sanitization is
```

File: docs/sanitation/sanitations.bal [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+                    schema.title = key.substring(1, key.length()) + "Wrapper";
+                    schema.title = schemaTitle;
+                } else if schemaTitle.endsWith("Type") {
+                    schema.title = schemaTitle.substring(0, schemaTitle.length() - 4);
+                }
+            }
+        }
+    }
+    check io:fileWriteJson(specPath, spec.toJson());
+};
```
Comment (niveathika) [inline docs/sanitation/sanitations.bal]:
```suggestion
};

```

**Review by niveathika — APPROVED** [review]

**Pull request merged**